src/System.IO.FileSystem/src/System/IO/Win32FileSystemEnumerable.cs

```diff
@@ -439,7 +440,6 @@ private void AddSearchableDirsToList(PathPair localSearchData)
                         // Setup search data for the sub directory and push it into the list
                         PathPair searchDataSubDir = new PathPair(tempUserPath, tempFullPath);
 
-                        Debug.Assert(_searchList != null, "_searchList should not be null");
```

Comment:
cc: @JeremyKuhne 
Moving this up to earlier, as by this point we've already dereferenced _searchList.